release.sh

```diff
@@ -1,25 +1,39 @@
 #!/usr/bin/env bash
-set -eux
+set -euo pipefail
 
-if [ -z "$1" ]
+if [ -z "${1:-}" ]
   then
-    echo "Missing version for master branch"
+    echo "Version needs to be provided."
+    echo "Usage: ./release.sh <version> <release-channel>"
     exit 1
 fi
 
-export $(grep -v '^\#' .env)
-if [ -z "$JETBRAINS_HUB_TOKEN" ]
+if [[ "${2:-}" != "stable" && "${2:-}" != "beta" ]]
+  then
+    echo "Release channel needs to be provided - supported values: stable, beta"
+    echo "Usage: ./release.sh <version> <release-channel>"
+    exit 1
+fi
+
+
+ENVS=$(grep -v '^\#' .env || true)
+if [[ -n "${ENVS}" ]]
+  then
+    export "${ENVS?}"
+fi
+
+if [ -z "${JETBRAINS_HUB_TOKEN:-}" ]
   then
     echo "Please put JETBRAINS_HUB_TOKEN in .env: JETBRAINS_HUB_TOKEN=<token>"
     exit 1
 fi
 
 NEW_VERSION=$1
 
-sed -i "s/<version>.*<\/version>/<version>$NEW_VERSION<\/version>/g" src/main/resources/META-INF/plugin.xml
+sed -i '' "s/<version>.*<\/version>/<version>$NEW_VERSION<\/version>/g" src/main/resources/META-INF/plugin.xml
```

Comment:
What do these two `''`actually do?

Reply:
With Mac the extension is compulsory: `-i extension`, and empty value disables the backup. However, empty value does not seem to work in sed bundled in Ubuntu, so I'll try to see what is best option here.